### PR TITLE
[chore] Upgrade react native example

### PR DESCRIPTION
**Pull request by fuller**
- Updates react-native to v0.57.8
- Updates Storybook to v4.1.6

This currently doesn't appear to be working with the latest version of Storybook, but I figured it'd be worthwhile to get the upgrade in place so it's easier to make it work again.

### CODE REVIEW

File: examples/react-native/android/app/build.gradle [hunk 1]
```diff
@@ -137,10 +137,10 @@ android {
...
 
 dependencies {
-    compile project(':react-native-devsettings-android')
-    compile fileTree(dir: "libs", include: ["*.jar"])
-    compile "com.android.support:appcompat-v7:23.0.1"
-    compile "com.facebook.react:react-native:+"  // From node_modules
+    implementation project(':react-native-devsettings-android')
+    implementation fileTree(dir: "libs", include: ["*.jar"])
+    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
+    implementation "com.facebook.react:react-native:+"  // From node_modules
```
Comment (wakeless) [inline examples/react-native/android/app/build.gradle]: :+1:

File: examples/react-native/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
-    android:versionName="1.0">
+    package="com.loki">
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
 
-    <uses-sdk
-        android:minSdkVersion="16"
-        android:targetSdkVersion="22" />
-
```
Comment (wakeless) [inline examples/react-native/android/app/src/main/AndroidManifest.xml]: :+1:

File: examples/react-native/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
     <application
       android:name=".MainApplication"
-      android:allowBackup="true"
```
Comment (wakeless) [inline examples/react-native/android/app/src/main/AndroidManifest.xml]: What does this do?

**Comment by wakeless** [comment]
I haven't compiled this but generally LGTM

**Comment by ajfuller** [comment]
Sorry I should have been more clear on the upgrade process I took. I used `react-native-git-upgrade`, which made a majority of the changes with few merge conflicts. The main merge conflict where I kept existing behavior was with `react-native-devsettings-android` https://github.com/oblador/loki/pull/126/files#diff-02f5b5162dec8a21bcff566ac215cc70R140, otherwise it should be equivalent to if we created a new project from scratch at this version.

**Comment by oblador** [comment]
Closing as current example is 0.61.

**Pull request closed without merging**